addon-test-support/@ember/test-helpers/setup-context.ts

```diff
@@ -132,6 +133,25 @@ export function resumeTest(): void {
   context.resumeTest();
 }
 
+const ORIGINAL_EMBER_ONERROR: (error: Error) => void | undefined = Ember.onerror;
+
+/**
+  Sets the Ember.onerror function for the duration of a single test. This
+  value is reset after each test to ensure correct test isolation.
+
+  @public
+  @param {Function} onError the onError function to be set on Ember.onerror
+*/
+export function setupOnerror(onError: (error: Error) => void): void {
+  if (!hasEmberVersion(2, 4)) {
```

Comment:
I think what we want is:

```js
let context = getContext();
let contextGuid = guidFor(context);
let contextCleanup = CLEANUP[contextGuid];

if (!Array.isArray(contextCleanup)) {
  throw new Error(
    'You must use `setupContext` / `setupRenderingContext` / `setupApplicationContext` in order to use `setupOnerror`'
  );
}
```

What do you think?

Reply:
You tell me. Is it less important the Ember version vs. the fact that this GUID is present in the CLEANUP?

Reply:
For this check, the Ember version is unrelated (Ember.onerror = function() {} has worked since Ember 1.0). The thing we care about is that you will get the ability to have it cleaned up for you...

Reply:
Yep. I understand. That seems fine. I'll change it.

Reply:
What should the corresponding conditional be around the test for throws? https://github.com/emberjs/ember-test-helpers/pull/540/files#diff-51eaa617037ec59cacc57f7ccd70ca8aR7

Reply:
I just left the test as is for now.